pkg/controller/namespace/deletion/namespaced_resources_deleter.go

```diff
@@ -135,10 +136,13 @@ func (d *namespacedResourcesDeleter) Delete(nsName string) error {
 	// there may still be content for us to remove
 	estimate, err := d.deleteAllContent(namespace.Name, *namespace.DeletionTimestamp)
 	if err != nil {
+		d.updateStatusCondition(namespace.Name, err)
```

Comment:
You need to clear this in cases where you had an error and then cleared that error.  Looking below, you can see that when this finishes, we issue an update, not a delete.